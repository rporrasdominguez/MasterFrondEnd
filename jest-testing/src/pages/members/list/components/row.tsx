import * as React from 'react';
import { Member } from '../viewModel';

interface Props {
  member: Member;
}

export const Row: React.StatelessComponent<Props> = (props) => (
  <tr className="item">
    <td>
      <img src={props.member.avatarUrl} style={{ maxWidth: '10rem' }} />
    </td>
    <td>
      <span>{props.member.id}</span>
    </td>
    <td>
      <span>{props.member.name}</span>
    </td>
  </tr>
);
