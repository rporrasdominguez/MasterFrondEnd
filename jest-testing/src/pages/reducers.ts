import { combineReducers } from 'redux';

export interface State {

}

export const reducers = combineReducers<State>({

});
