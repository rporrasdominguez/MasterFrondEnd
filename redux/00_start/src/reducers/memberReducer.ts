import {actionsEnums} from '../common/actionsEnums';
import {MemberEntity} from '../model/member';

export type memberState = {
  members: MemberEntity[];
  company:string;
}

interface BaseAction{
  type:string;
  payload:any;
}

const createDefaultState = (): memberState =>({
  members: [],
  company: 'lemoncode',
})

export const memberReducer =  (state : memberState = [], action) => {
  switch (action.type) {
    case actionsEnums.MEMBER_REQUEST_COMPLETED:
      return handleMemberRequestCompletedAction(state, action.payload);
  }

  return state;
};

const handleMemberRequestCompletedAction = (state : memberState, members) => {
  return members;
}
