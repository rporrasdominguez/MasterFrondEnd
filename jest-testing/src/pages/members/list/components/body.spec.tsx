import * as React from 'react';
import { Body } from './body';
import { Row } from './row';

import { shallow, mount } from 'enzyme';
import { membersReducer } from '../reducers';

describe('pages/members/list/components/body specs', () => {
  it('should render as expected when passing one item', () => {

    // Arrange
    const props ={
      members:[
          {
              id: 1,
              name: 'name test',
              avatarUrl: 'url test',
          }
        ]    
    }
    // Act
    const component = mount(<Body {...props}/>);
 
    

    // Assert
    expect(component).toMatchSnapshot();
  });

  it('should render as expected when passing more than one item ', () => {

    // Arrange
    const props = {
      members:[
        {
          id: 1,
          name: 'name test',
          avatarUrl: 'url test',
        } ,
        {
          id: 2,
          name: 'name test 2',
          avatarUrl: 'url test 2',
       } 
      ]
    };

    // Act
    const component = mount(<Body {...props}/>);


    // Assert
    expect(component).toMatchSnapshot();
   
  });



});