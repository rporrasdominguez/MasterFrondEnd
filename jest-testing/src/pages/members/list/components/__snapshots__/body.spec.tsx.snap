// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/members/list/components/body specs should render as expected when passing more than one item  1`] = `
<Component
  members={
    Array [
      Object {
        "avatarUrl": "url test",
        "id": 1,
        "name": "name test",
      },
      Object {
        "avatarUrl": "url test 2",
        "id": 2,
        "name": "name test 2",
      },
    ]
  }
>
  <tbody
    className="list-items"
  >
    <Component
      key="1"
      member={
        Object {
          "avatarUrl": "url test",
          "id": 1,
          "name": "name test",
        }
      }
    >
      <tr
        className="item"
      >
        <td>
          <img
            src="url test"
            style={
              Object {
                "maxWidth": "10rem",
              }
            }
          />
        </td>
        <td>
          <span>
            1
          </span>
        </td>
        <td>
          <span>
            name test
          </span>
        </td>
      </tr>
    </Component>
    <Component
      key="2"
      member={
        Object {
          "avatarUrl": "url test 2",
          "id": 2,
          "name": "name test 2",
        }
      }
    >
      <tr
        className="item"
      >
        <td>
          <img
            src="url test 2"
            style={
              Object {
                "maxWidth": "10rem",
              }
            }
          />
        </td>
        <td>
          <span>
            2
          </span>
        </td>
        <td>
          <span>
            name test 2
          </span>
        </td>
      </tr>
    </Component>
  </tbody>
</Component>
`;

exports[`pages/members/list/components/body specs should render as expected when passing one item 1`] = `
<Component
  members={
    Array [
      Object {
        "avatarUrl": "url test",
        "id": 1,
        "name": "name test",
      },
    ]
  }
>
  <tbody
    className="list-items"
  >
    <Component
      key="1"
      member={
        Object {
          "avatarUrl": "url test",
          "id": 1,
          "name": "name test",
        }
      }
    >
      <tr
        className="item"
      >
        <td>
          <img
            src="url test"
            style={
              Object {
                "maxWidth": "10rem",
              }
            }
          />
        </td>
        <td>
          <span>
            1
          </span>
        </td>
        <td>
          <span>
            name test
          </span>
        </td>
      </tr>
    </Component>
  </tbody>
</Component>
`;
